### PR TITLE
Example: mTLS between RabbitMQ nodes

### PR DESCRIPTION
**Pull request by mkuratczyk**
This example shows how to configure mTLS between RabbitMQ nodes. It uses Cert Manager but could be used without it (secret would have to be created manually).

### CODE REVIEW

File: docs/examples/mtls-inter-node/rabbitmq.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+apiVersion: rabbitmq.com/v1beta1
+kind: RabbitmqCluster
+metadata:
+  name: mtls-inter-node
+spec:
+  rabbitmq:
+    envConfig: |
+      ERL_SSL_PATH="/usr/local/lib/erlang/lib/ssl-10.1/ebin"
+      SERVER_ADDITIONAL_ERL_ARGS="-pa /usr/local/lib/erlang/lib/ssl-10.1/ebin -proto_dist inet_tls -ssl_dist_optfile /etc/rabbitmq/inter-node-tls.config"
```
Comment (lukebakken) [inline docs/examples/mtls-inter-node/rabbitmq.yaml:9]: If you specify the ssl path here and in the next variable, then `ERL_SSL_PATH` does not need to be set.

**Review by MirahImage — APPROVED** [review]

**Review by ferozjilla — APPROVED** [review]

**Comment by ferozjilla** [comment]
It all works for me. 
- Minor suggestions for grammar
- Adding links where possible
- Asking some small questions for understanding.

File: docs/examples/mtls-inter-node/rabbitmq-certificate.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    algorithm: RSA
+    encoding: PKCS1
+    size: 2048
+  usages:
+    - server auth
+    - client auth
+  dnsNames:
+  - mtls-inter-node-server-0.mtls-inter-node-nodes.default
+  - mtls-inter-node-server-1.mtls-inter-node-nodes.default
+  - mtls-inter-node-server-2.mtls-inter-node-nodes.default
```
[the remark this answers is not in the document]
Comment (mkuratczyk) [reply docs/examples/mtls-inter-node/rabbitmq-certificate.yaml:24]: Yes, I tested the wildcard option shortly after submitting the PR. Wildcard would make it easier to add nodes to the cluster for example, but I know some companies don't like wildcard certs.

File: docs/examples/mtls-inter-node/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+* `rabbitmq-certificate.yaml` - defines a certificate that will be provisioned by Cert Manager and then mounted as a volume
+
+`setup.sh` should perform all the necessary steps but may need to be adjusted to work on your system.
+
+```shell
+# install Cert Manager
+kubectl apply --validate=false -f https://github.com/jetstack/cert-manager/releases/download/v1.0.4/cert-manager.yaml
+# deploy the example
+./setup.sh
+```
```
[the remark this answers is not in the document]
Comment (mkuratczyk) [reply docs/examples/mtls-inter-node/README.md:23]: We can always add it :)

File: docs/examples/mtls-inter-node/inter_node_tls.config [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  {client, [
+    {cacertfile, "/etc/rabbitmq/certs/ca.crt"},
+    {certfile,   "/etc/rabbitmq/certs/tls.crt"},
+    {keyfile,    "/etc/rabbitmq/certs/tls.key"},
+    {secure_renegotiate, true},
+    {fail_if_no_peer_cert, true},
+    {verify, verify_peer},
+    {customize_hostname_check, [
+      {match_fun, public_key:pkix_verify_hostname_match_fun(https)}
+    ]}
```
[the remark this answers is not in the document]
Comment (mkuratczyk) [reply docs/examples/mtls-inter-node/inter_node_tls.config:22]: Taken from @lukebakken's example (https://github.com/lukebakken/erlang-inet-dist). As far as I tested, without this, Erlang would accept any name in the certificate.

**Comment by MirahImage** [comment]
Thanks for adding the commands for verifying the TLS connection

**Pull request merged**